erpnext/stock/doctype/item/item.py

```diff
@@ -898,6 +905,72 @@ def validate_auto_reorder_enabled_in_stock_settings(self):
 	def validate_valuation_method(self):
 		if self.is_new() and not self.valuation_method:
 			self.valuation_method = frappe.db.get_single_value("Stock Settings", "valuation_method")
+	
+	def update_to_bloomtrace(self):
+		"""
+		Create a new item on Bloomtrace via Bloomstack.
+		Args: 
+		
+		Returns: 
+			Successful Message to the User
+		"""
+		try:
+			data = self.as_dict()
+			request_data = {
+				"site_url": "manufacturing.bloomstack.io",
+				"customer_name": "Bloomstack",
+				"company_name": "Bloomstack India",
+				"license_number": "A12-0000015-LIC",
+				"rest_method": "POST",
+				"environment": "sandbox",
+				"doctype": self.doctype,
+				"doctype_data": data
+			}
+
+			# to map the metrc_item_name, metrc_uom, metrc_item_category
+			request_data['doctype_data']['name'] = self.metrc_item_name
+			request_data['doctype_data']['uom'] = self.metrc_uom
+			request_data['doctype_data']['item_category'] = self.metrc_item_category
+
+			# passing the hard coded modified_by
+			request_data['doctype_data']['modified_by'] = "neil@bloomstack.com"
+
+			# for item_category which requires strain
+			if self.strain_name:
+				request_data['doctype_data']['strain'] = self.strain_name
+				
+			# getting the item category from the Compliance Item Category for quantity type and mandatory unit
+			item_category = frappe.db.get("Compliance Item Category", self.metrc_item_category, ['quantity_type', 'mandatory_unit'])
+
+			# setting the unit uom and unit value as per the countbased quantity type
+			if item_category.get('quantity_type') == "CountBased":
+				if item_category.get('mandatory_unit') == "Volume": 
+					request_data['doctype_data']['unit_volume'] = self.metrc_unit_value
+					request_data['doctype_data']['unit_volume_uom'] = self.metrc_unit_uom
+				if item_category.get('mandatory_unit') == "Weight": 
+					request_data['doctype_data']['unit_weight'] = self.metrc_unit_value
+					request_data['doctype_data']['unit_weight_uom'] = self.metrc_unit_uom
+
+			# create am item on the bloomtrace
+			bloomtrace_response = requests.post('https://bl2qu9obqb.execute-api.ap-south-1.amazonaws.com/dev/doctype/create-item', json=request_data)
```

Comment:
same, url needs to come from bloomtrace settings

Reply:
raise a PR for the settings and mark this PR as dependent on that

Reply:
yeah, raising a PR for bloomtrace settings